hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1725,6 +1729,173 @@ public void delete(final Path path, final boolean recursive,
     } while (shouldContinue);
   }
 
+  private void deleteBlobPath(final Path path,
+      final boolean recursive,
+      final TracingContext tracingContext) throws IOException {
+    StringBuilder listSrcBuilder = new StringBuilder();
+    listSrcBuilder.append(path.toUri().getPath());
+    if (!path.isRoot()) {
+      listSrcBuilder.append(FORWARD_SLASH);
+    }
+
+    String listSrc = listSrcBuilder.toString();
+    BlobList blobList = client.getListBlobs(null, listSrc, null, null,
+        tracingContext).getResult().getBlobList();
+    if (blobList.getBlobPropertyList().size() > 0) {
+      LOG.debug(
+          String.format("Path %s is child-blobs", path.toUri().getPath()));
+      if (!recursive) {
+        LOG.error(
+            String.format("Non-recursive delete of non-empty directory %s",
+                path.toUri().getPath()));
+        throw new IOException(
+            "Non-recursive delete of non-empty directory");
+      }
+      ListBlobQueue queue = new ListBlobQueue(blobList.getBlobPropertyList(),
+          getAbfsConfiguration().getProducerQueueMaxSize(),
+          getAbfsConfiguration().getBlobDirDeleteMaxThread());
+      if (blobList.getNextMarker() != null) {
+        new ListBlobProducer(listSrc, client, queue,
+            blobList.getNextMarker(), tracingContext);
+      } else {
+        queue.complete();
+      }
+      ListBlobConsumer consumer = new ListBlobConsumer(queue);
+      deleteCheckOnParentBlob(path, tracingContext);
+      deleteOnConsumedBlobs(path, consumer, tracingContext);
+      return;
+    }
+    LOG.debug(String.format("Path %s doesn't have child-blobs",
+        path.toUri().getPath()));
+    if (!path.isRoot()) {
+      try {
+        getBlobProperty(path, tracingContext);
+      } catch (AbfsRestOperationException ex) {
+        if (ex.getStatusCode() == HTTP_NOT_FOUND) {
```

Comment:
Do we not need to throw other exceptions ?

Reply:
Correct!